uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java

```diff
@@ -0,0 +1,196 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.broker;
+
+import androidx.annotation.NonNull;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+import java.util.concurrent.TimeUnit;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * An abstract representation of a Broker Host App. This class contains all the common
+ * functionality that can be used to interact with a Broker Host App during UI Test.
+ */
+abstract class AbstractBrokerHost {
+    public final static String BROKER_HOST_APP_PACKAGE_NAME = "com.microsoft.identity.testuserapp";
+    public final static String BROKER_HOST_APP_NAME = "Broker Host App";
+    private final static long APP_LAUNCH_TIMEOUT = TimeUnit.SECONDS.toMillis(5);
+    // Resource id's
+    private final static String HEADER_RESOURCE_ID = "text_header";
+    protected final static String USERNAME_EDIT_TEXT = "edit_text_username";
+    protected final static String DIALOG_BOX_RESOURCE_ID = "android:id/message";
+    protected final static String DIALOG_BOX_OK_BUTTON_RESOURCE_ID = "android:id/button1";
+
+    @AllArgsConstructor
+    protected enum BrokerHostNavigationMenuItem {
+        MULTIPLE_WPJ_API("multiple_wpj_api", "Multiple WPJ API"),
+        SINGLE_WPJ_API("single_wpj_api", "Single WPJ API"),
+        BROKER_API("broker_api", "Broker API"),
+        FLIGHTS_API("broker_flights", "Broker Flights API");
+
+        @Getter
+        private final String resourceId;
+        @Getter
+        private final String title;
+    }
+
+    /**
+     * This method clicks on the button with the given resource id
+     *
+     * @param resourceIdButton the resource id of the button to be clicked
+     */
+    public void clickButton(@NonNull final String resourceIdButton) {
+        final String resourceId = CommonUtils.getResourceId(
+                BROKER_HOST_APP_PACKAGE_NAME,
+                resourceIdButton
+        );
+        UiAutomatorUtils.handleButtonClick(resourceId);
+    }
+
+    /**
+     * This method fills the text box with the given resource id with the given text
+     *
+     * @param resourceEditText the resource id of the text box
+     * @param text             the text to be write in the text box
+     */
+    public void fillTextBox(@NonNull final String resourceEditText, @NonNull final String text) {
```

Comment:
Static?

Reply:
Consider this for all of these methods handling UI

Reply:
done